flask/app.py

```diff
@@ -1370,6 +1390,33 @@ def url_defaults(self, f):
         self.url_default_functions.setdefault(None, []).append(f)
         return f
 
+    def _find_error_handler(self, e):
+        """Finds a registered error handler for the request’s blueprint.
+        If neither blueprint nor App has a suitable handler registered, returns None
+        """
+        exc_class, code = self._get_exc_class_and_code(type(e))
+        
+        def find_superclass(d):
+            if not d:
+                return None
+            for superclass in exc_class.mro():
+                if superclass is BaseException:
+                    return None
+                handler = d.get(superclass)
+                if handler is not None:
+                    d[exc_class] = handler  # cache for next time exc_class is raised
```

Comment:
This might be confusing during debugging, since the handler dict mutates during execution, but you might argue that when debugging you should be aware of the cache, so I don't feel strongly about this, I just don't like premature optimization.